loki/transformations/tests/test_data_offload.py

```diff
@@ -805,3 +859,457 @@ def test_transformation_global_var_derived_type_hoist(here, config, frontend, ho
     assert kernel.variable_map['p'].type.dtype.name == 'point'
     assert kernel.variable_map['p0'].type.intent == 'in'
     assert kernel.variable_map['p0'].type.dtype.name == 'point'
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_field_offload(frontend, parkind_mod, field_module, tmp_path):
+    fcode = """
+    module driver_mod
+      use parkind1, only: jprb
+      use field_module, only: field_2rb, field_3rb
+      implicit none
+
+      type state_type
+        real(kind=jprb), dimension(10,10), pointer :: a, b, c
+        class(field_3rb), pointer :: f_a, f_b, f_c
+        contains
+        procedure :: update_view => state_update_view
+      end type state_type
+
+    contains
+
+      subroutine state_update_view(self, idx)
+        class(state_type), intent(in) :: self
+        integer, intent(in)           :: idx
+      end subroutine
+
+      subroutine kernel_routine(nlon, nlev, a, b, c)
+        integer, intent(in)             :: nlon, nlev
+        real(kind=jprb), intent(in)     :: a(nlon,nlev)
+        real(kind=jprb), intent(inout)  :: b(nlon,nlev)
+        real(kind=jprb), intent(out)    :: c(nlon,nlev)
+        integer :: i, j
+
+        do j=1, nlon
+          do i=1, nlev
+            b(i,j) = a(i,j) + 0.1
+            c(i,j) = 0.1
+          end do
+        end do
+      end subroutine kernel_routine
+
+      subroutine driver_routine(nlon, nlev, state)
+        integer, intent(in)             :: nlon, nlev
+        type(state_type), intent(inout) :: state
+        integer                         :: i
+
+        !$loki data
+        do i=1,nlev
+            call state%update_view(i)
+            call kernel_routine(nlon, nlev, state%a, state%b, state%c)
+        end do
+        !$loki end data
+
+      end subroutine driver_routine
+    end module driver_mod
+    """
+    driver_mod = Sourcefile.from_source(fcode, frontend=frontend, xmods=[tmp_path])['driver_mod']
+    driver = driver_mod['driver_routine']
+    deviceptr_prefix = 'loki_devptr_prefix_'
+    driver.apply(FieldOffloadTransformation(devptr_prefix=deviceptr_prefix,
+                                            offload_index='i',
+                                            field_group_types=['state_type']),
+                 role='driver',
+                 targets=['kernel_routine'])
+
+    calls = FindNodes(CallStatement).visit(driver.body)
+    kernel_call = next(c for c in calls if c.name=='kernel_routine')
+
+    # verify that field offloads are generated properly
+    in_calls = [c for c in calls if 'get_device_data_rdonly' in c.name.name.lower()]
```

Comment:
Is case-sensitive comparison really needed here? Same goes for a bunch of lines below.

Reply:
Yes, actually the tests fails without case-sensitive comparison, because the generated ``ProcedureSymbol``s actually have uppercase names.

Reply:
Ahhh, I see, and they have parents, so `==` comparison doesn't work. Yeah, ok, makes sense then. Please leave as is.